spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/r2dbc/ConnectionFactoryBuilder.java

```diff
@@ -16,17 +16,21 @@
 
 package org.springframework.boot.autoconfigure.r2dbc;
 
+import java.util.List;
 import java.util.function.Consumer;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
+import io.r2dbc.proxy.ProxyConnectionFactoryProvider;
+import io.r2dbc.proxy.listener.ProxyExecutionListener;
```

Comment:
We can't really do that, can we? `ConnectionFactoryBuilder` should only rely on the SPI.

Reply:
ok, I made not to throw `NoClassDefFoundError` but I'll go some other way not to use them here.